Tactics/ArmConstr.lean

```diff
@@ -0,0 +1,392 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Shilpi Goel
+-/
+
+/-
+Experimental method to aggregate state effects using reflection. This is
+inspired by `simp_arith`, especially the following files:
+
+`[lean] Init/Data/Nat/Linear.lean`
+`[lean] Meta/Tactic/LinearAith/Nat/Basic.lean`
+`[lean] Meta/Tactic/LinearAith/Nat/Simp.lean`
+-/
+
+import Arm.Exec
+
+namespace ArmConstr
+
+/- We use `Nat`s to refer to all the state variables in our context. -/
+abbrev StateVar := Nat
+
+/-- A `GPRVal` can either be a variable or a read from the previous state.
+
+  This is very barebones right now -- for instance, we'd need to support
+  `BitVec` operators here.
+-/
+inductive GPRVal where
+  | var (i : Nat)
+  | r_gpr (i : BitVec 5)
+  deriving DecidableEq, Repr, Inhabited
+
+abbrev StateContext := List ArmState
+abbrev GPRValContext := List (BitVec 64)
```

Comment:
I find the name a bit confusing, and I'm also not sure why these need to be looked up. Can't they directly be `(BitVec 64)`s that are reflected?